indispensable/src/commonMain/kotlin/at/asitplus/signum/indispensable/asn1/Asn1Decoding.kt

```diff
@@ -298,27 +309,134 @@ fun Long.Companion.decodeFromDer(bytes: ByteArray): Long = runRethrowing {
 @Throws(Asn1Exception::class)
 private fun ByteArray.readTlv(): TLV = runRethrowing {
     if (this.isEmpty()) throw IllegalArgumentException("Can't read TLV, input empty")
-    val tag = this[0].toUByte()
-    if (this.size == 1) return TLV(tag, byteArrayOf())
-    val firstLength = this[1]
-    if (firstLength == 0x82.toByte()) {
-        if (this.size < 4) throw IllegalArgumentException("Can't decode length")
-        val length = (this[2].toUByte().toInt() shl 8) + this[3].toUByte().toInt()
-        if (this.size < 4 + length) throw IllegalArgumentException("Out of bytes")
-        val value = this.drop(4).take(length).toByteArray()
-        return TLV(tag, value)
+    if (this.size == 1) return TLV(Asn1Element.Tag(byteArrayOf(this[0])), byteArrayOf())
+
+    val decodedTag = decodeTag()
+    val tagLength = decodedTag.second.size
+    val tagBytes = decodedTag.second
+    val value = drop(tagLength).decodeValue()
```

Comment:
@nodh this is one for you

Reply:
Is this better? https://github.com/a-sit-plus/signum/blob/feature/asn1ProperTagging-tlv/indispensable/src/commonMain/kotlin/at/asitplus/signum/indispensable/asn1/Asn1Decoding.kt#L319

Reply:
@iaik-jheher your call

Reply:
i like it better, though i'd still prefer if we just stored a bytearray view here instead of a full bytearray (as arg2 to tag)

the underlying issue where we're constantly copying data into ever-smaller bytearrays still persists

Reply:
i'll merge it into this branch, and pin this to #120